packages/ui/src/components/ResourceBlock/ResourceBlock.tsx

```diff
@@ -0,0 +1,197 @@
+      data-h2-background-color="base(foreground)"
+    >
+      {/* heading bar */}
+      <div
+        {...headingBarStyleMap[headingColor]}
+        data-h2-border-radius="base(rounded rounded 0 0)"
+        data-h2-border-bottom-width="base(1px)"
+        data-h2-border-bottom-style="base(solid)"
+        data-h2-padding="base(x1) p-tablet(x1 x1.5 x1 x1.5)"
+        data-h2-text-align="base(center)"
+      >
+        {/* wrapper */}
+        <div {...wrapperStyleMap[headingColor]}>
+          <HeadingTextElement
+            {...commonHeadingTextStyles}
+            data-h2-text-align="base(center) p-tablet(left)"
+          >
+            {title}
+          </HeadingTextElement>
+        </div>
+      </div>
+      {/* content */}
+      <div data-h2-display="base(flex)" data-h2-flex-direction="base(column)">
+        {children}
+      </div>
+    </div>
+  );
+};
+
+interface ItemProps {
+  link: ReactElement<LinkProps>;
+  description: string;
+  state?: "incomplete" | "complete";
+}
+
+const getStateIcon = (state: ItemProps["state"]): ReactNode | null => {
+  const commonStyles: HydrogenAttributes = {
+    "data-h2-width": "base(20px)",
+    "data-h2-height": "base(20px)",
+    "data-h2-position": "base(absolute)",
+    "data-h2-location": "base(x0.75, x0.75, auto, auto)",
+  };
+
+  if (state === "incomplete") {
+    return (
+      <ExclamationCircleIcon data-h2-color="base(error)" {...commonStyles} />
+    );
+  }
+  if (state === "complete") {
+    return <CheckCircleIcon data-h2-color="base(success)" {...commonStyles} />;
+  }
+  return null;
+};
+
+const Item = ({ link: rawLink, description, state }: ItemProps) => {
+  // prepare link
+  const link = rawLink
+    ? cloneElement(rawLink, {
+        color: "black",
+        // yuck, style exception 😞
+        "data-h2-font-weight": "base(bold)",
+        // big click target black magic 🧙
+        "data-h2-position": "base:selectors[::after](absolute)",
+        "data-h2-content": "base:selectors[::after](' ')",
+        "data-h2-inset": "base:selectors[::after](0)",
+        "data-h2-justify-self": "base(end)",
+      })
+    : null;
+
+  const extraStateStyles =
+    state === "incomplete"
+      ? {
+          // should match the absolute positioning of the state icon ( + half the height & width to get the center)
+          "data-h2-background":
+            "base(radial-gradient(circle x5 at top calc(x0.75 + 10px) right calc(x0.75 + 10px), error.10, foreground))",
+        }
+      : {};
+
+  return (
+    <div
+      data-h2-background="base(foreground)"
+      data-h2-padding="base(x1)"
+      data-h2-display="base(flex)"
+      data-h2-flex-direction="base(column)"
+      data-h2-gap="base(x0.15)"
+      data-h2-border-bottom="base:selectors[:not(:last-child)](1px solid gray.lighter)"
+      // make the containing block for status icon
+      data-h2-position="base(relative)"
+      {...extraStateStyles}
+    >
+      {getStateIcon(state)}
```

Comment:
`{getStateIcon(state)}` could be refactored to it's own component within this file: `<StateIcon state={state} />`.

Reply:
Good idea!
[refactor state icon](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11200/commits/796512574726a40ef07027a78c68432fd56e7802)